libraries/ESP8266WebServer/src/ESP8266WebServer-impl.h

```diff
@@ -420,7 +420,8 @@ void ESP8266WebServerTemplate<ServerType>::stop() {
 }
 
 template <typename ServerType>
-void ESP8266WebServerTemplate<ServerType>::sendHeader(String&& name, String&& value, bool first) {
+template <typename S1, typename S2>
+void ESP8266WebServerTemplate<ServerType>::sendHeader(S1 name, S2 value, bool first) {
```

Comment:
minor nit

at a glance, we have a bunch of typed pairs that generate multiple emplace\_... for every sendHeader variant
I'd consider using perfect forward and a specific type we already have

```cpp
auto foo(S1&& name, S2&& value, bool condition) {
    auto pair = std::make_pair(
        String(std::forward<S1>(name)),
        String(std::forward<S2>(value)));

    if (condition) {
        foo(std::move(pair));
    } else {
        bar(std::move(pair));
    }
}
```